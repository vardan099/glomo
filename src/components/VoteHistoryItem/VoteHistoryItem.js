import React from 'react';
import PropTypes from "prop-types";
const voteTypes = {
    1: {title:"Home Team",value:"home"},
    2: {title:"Draw",value:"draw"},
    3: {title:"Away Team",value:"away"},
};

const VoteHistoryItem = (props) => {
    const {poll, voteType} =  props;
    return(
        <div className="history-item w-100">
            <div>
                <p>{`${poll.homeName} - ${poll.awayName}`}</p>
            </div>
            <div>
                <p>{poll.group} - {poll.sport}</p>
            </div>
            <div>
                <div className={`bg-${voteTypes[voteType].value} history-item_vote-type`}>
                    <span>{voteTypes[voteType].title}</span>
                </div>
            </div>
        </div>
    )
};

VoteHistoryItem.propTypes = {
    poll: PropTypes.object,
    voteType: PropTypes.number
};
export default VoteHistoryItem