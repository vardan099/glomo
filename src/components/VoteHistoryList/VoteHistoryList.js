import React, {useState, useEffect} from 'react';
import PropTypes from "prop-types";
import VoteHistoryItem from '../VoteHistoryItem/VoteHistoryItem'

const VoteHistoryList = (props) => {
    const {polls, history} =  props;
    const [voteList,setVoteList] = useState([]);
    const [showAll,setShowAll] = useState(false);

    useEffect(() => {
        let voteList;
        if(showAll){
            voteList = [...history].reverse();
        }else{
            voteList = history.slice(-5).reverse();
        }
        setVoteList(voteList);
    }, [history, showAll]);


    const handleSeeAllClick = () =>{
        setShowAll(!showAll)
    };
    return(
        <div className="history w-100">
            <div className="history_title">
                <h3>History</h3>
                <p className="history_title-info">Count of votes: <strong>{history.length}</strong><span className='history_title-switcher' onClick={handleSeeAllClick}>{showAll?'See less':'See all'}</span></p>
            </div>
            {history.length > 0 && Object.entries(polls).length > 0 ?
                <div className="history_list custom-scroll">
                    {voteList && voteList.map((item,key) => (
                        <VoteHistoryItem key={key} poll={polls[item.pollId]} voteType={item.voteType}/>
                    ))}
                </div> :
                <h3 className="history_no-votes">No votes to show</h3>
            }
        </div>
    )
};

VoteHistoryList.propTypes = {
    polls: PropTypes.object,
    history: PropTypes.array
};
export default VoteHistoryList