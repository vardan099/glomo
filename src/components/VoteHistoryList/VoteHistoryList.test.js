import React from 'react';
import { shallow } from 'enzyme';
import { checkProps } from '../../TestUtils/index';
import VoteHistoryList from './VoteHistoryList';
import VoteHistoryItem from "../VoteHistoryItem/VoteHistoryItem";
describe('VoteHistoryList Component', () => {

    describe('Checking PropTypes', () => {
        it('Should NOT throw a warning', () => {
            const expectedProps = {
                polls: jest.object,
                history: jest.array
            };
            const propsError = checkProps(VoteHistoryList, expectedProps);
            expect(propsError).toBeUndefined();
        });
    });

    describe('Component Renders', () => {
        let wrapper;
        beforeEach(() => {
            const polls = {
                JxrZyQKTrw: {
                    awayName: "Panthrakikos Komotini",
                    createdAt: "2015-12-18T12:30:39.228Z",
                    group: "Greek Cup",
                    homeName: "Chania FC",
                    id: 1002916450,
                    name: "Chania FC - Panthrakikos Komotini",
                    objectId: "1UaQjc7lIb",
                    sport: "FOOTBALL",
                    country: "ENGLAND",
                    state: "STARTED",
                },
                UPJ240T2Qj: {
                    awayName: "PAOK Thessaloniki",
                    createdAt: "2015-12-18T12:30:39.234Z",
                    group: "Greek Cup",
                    homeName: "Olympiakos Volos",
                    id: 1002916451,
                    name: "Olympiakos Volos - PAOK Thessaloniki",
                    objectId: "UPJ240T2Qj",
                    sport: "FOOTBALL",
                    country: "FRANCE",
                    state: "STARTED",
                }
            };

            const props ={
                polls,
                history: [
                    {"pollId":"UPJ240T2Qj","voteType":3,"createdAt":1578228998690},
                    {"pollId":"JxrZyQKTrw","voteType":1,"createdAt":1578229000433}
                    ]
            };
            wrapper = shallow(<VoteHistoryList {...props} />);
        });

        it('renders correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

    });






});