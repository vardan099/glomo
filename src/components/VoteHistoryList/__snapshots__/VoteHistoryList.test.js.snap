// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteHistoryList Component Component Renders renders correctly 1`] = `ShallowWrapper {}`;
